src/v2/api/services/PermissionService.ts

```diff
@@ -22,22 +22,21 @@ export class PermissionService {
     return this.checkPermission(roles, permission);
   }
 
-  hasRolePermission (grants: Grant[], permission: string) {
-    for (const grant of grants) {
-      const hasPermission = this.hasGrantPermission(permission, grant.permission);
-      if (hasPermission) return grant.set;
-    }
+  hasRolePermission (grants: Grant[], permission: string): boolean | undefined {
+    const grant = grants
+      .find((grant) => grant.permission && this.hasGrantPermission(permission, grant.permission));
+
+    return grant?.set;
   }
 
-  hasGrantPermission (permission: string, grant: string) {
+  hasGrantPermission (permission: string, grant: string): boolean {
     const parts = permission.split('.');
     const grantParts = grant.split('.');
 
     if (grantParts.length > parts.length) return false;
 
     for (let i = 0; i < parts.length; i++) {
       const part = grantParts[i];
-      if (!part) return false;
       if (part === '*') {
         if (i === grantParts.length - 1) return true;
         else continue;
```

Comment:
There is no need to use continue, just use else-if operator. It will be more obvious codeflow.